rir/src/interpreter/interp.cpp

```diff
@@ -491,6 +491,264 @@ RIR_INLINE SEXP legacyCall(const CallContext& call, Context* ctx) {
+        if (XLENGTH(arg) != 1)
+            return nullptr;
+        return TYPEOF(arg) == VECSXP ? R_TrueValue : R_FalseValue;
+    }
+
+    case 395: { // "is.na"
+        if (nargs != 1)
+            return nullptr;
+
+        auto arg = args[0];
+        if (XLENGTH(arg) != 1)
+            return nullptr;
+
+        switch (TYPEOF(arg)) {
+        case INTSXP:
+            return INTEGER(arg)[0] == NA_INTEGER ? R_TrueValue : R_FalseValue;
+        case LGLSXP:
+            return LOGICAL(arg)[0] == NA_LOGICAL ? R_TrueValue : R_FalseValue;
+        case REALSXP:
+            return ISNAN(REAL(arg)[0]) ? R_TrueValue : R_FalseValue;
+        default:
+            return nullptr;
+        }
+        assert(false);
+    }
+
+    case 393: { // "is.function"
+        if (nargs != 1)
+            return nullptr;
+        return isFunction(args[0]) ? R_TrueValue : R_FalseValue;
+    }
+
+    case 507: { // "islistfactor"
+        if (nargs != 2)
+            return nullptr;
+        auto n = XLENGTH(args[0]);
+        if (n == 0 || !isVectorList(args[0]))
+            return R_FalseValue;
+        int recursive = asLogical(args[1]);
+        if (recursive)
+            return nullptr;
+
+        for (int i = 0; i < n; i++)
+            if (!isFactor(VECTOR_ELT(args[0], i)))
+                return R_FalseValue;
+
+        return R_TrueValue;
+    }
+
+    case 88: { // "length"
+        if (nargs != 1)
+            return nullptr;
+
+        switch (TYPEOF(args[0])) {
+        case INTSXP:
+        case REALSXP:
+        case LGLSXP:
+            return Rf_ScalarInteger(XLENGTH(args[0]));
+        default:
+            return nullptr;
+        }
+        assert(false);
+    }
+
+    case 386: { // "is.numeric"
+        if (nargs != 1)
+            return nullptr;
+        return isNumeric(args[0]) && !isLogical(args[0]) ? R_TrueValue
+                                                         : R_FalseValue;
+    }
+
+    case 387: { // "is.matrix"
+        if (nargs != 1)
+            return nullptr;
+        return isMatrix(args[0]) ? R_TrueValue : R_FalseValue;
+    }
+
+    case 388: { // "is.array"
+        if (nargs != 1)
+            return nullptr;
+        return isArray(args[0]) ? R_TrueValue : R_FalseValue;
+    }
+
+    case 389: { // "is.atomic" (389) nargs : 1 arg0 : double a 0
+        if (nargs != 1)
+            return nullptr;
+        switch (TYPEOF(args[0])) {
+        case NILSXP:
+            /* NULL is atomic (S compatibly), but not in isVectorAtomic(.) */
+        case CHARSXP:
+        case LGLSXP:
+        case INTSXP:
+        case REALSXP:
+        case CPLXSXP:
+        case STRSXP:
+        case RAWSXP:
+            return R_TrueValue;
+        default:
+            return R_FalseValue;
+        }
+        assert(false);
+    }
+
+    case 384: { // "is.object" (384) nargs : 1 arg0 : double a 0
+        if (nargs != 1)
+            return nullptr;
+        return OBJECT(args[0]) ? R_TrueValue : R_FalseValue;
```

Comment:
What about `isObject`?

Reply:
This is what the builtin does... straight copy from the R code. Don't ask me why it's not using isObject :)